test-skripsi/test-1.py

```diff
@@ -0,0 +1,41 @@
+
+def calculate_discount(price, discount):
+    discounted_price = price * discount
+    return discounted_price
+
+def factorial(n):
+    if n == 1:
+        return 1
+    else:
+        return n * factorial(n)  
+
+def parse_user_input():
+    try:
+        user_input = input("Enter a number: ")
+        number = int(user_input)
+    except ValueError:
+        print("Invalid input! Please enter a valid number.")
+        return None
+    return number
+
+def fibonacci(n):
+    fib_sequence = [0, 1]
```

Comment:
For the `fibonacci` function, the initial sequence `[0, 1]` is correct, but the function's logic could be optimized. For example, storing the entire sequence is not necessary for calculating the nth Fibonacci number. Consider using a more space-efficient approach that only keeps track of the last two numbers.